### ISSUE TITLE
Where does hook_logging_tensor log?

### ISSUE THREAD

**Issue by adder** [rstudio/tfestimators#99]
Hey,
I'm a bit at a loss how `hook_logging_tensor` works.
I'm using it as described in the example : https://github.com/rstudio/tfestimators/blob/master/vignettes/examples/tensorflow_layers.R
But I don't see any logging info printed or written to a file. Where should I look and how should I use this hook?

Best

**Comment by terrytangyuan** [comment]
It will only log during prediction phase https://github.com/rstudio/tfestimators/blob/master/vignettes/examples/tensorflow_layers.R#L69

I forgot to include a predict() call to this example. I will add that tomorrow but could you try that and see if it logs that tensor?

**Comment by adder** [comment]
You're right, thanks!
When it put the hook in `predict()` something gets printed to the r interpreter.
So to get things straight: 
1) It's only possible to use `hook_logging_tensor` with `predict()`? I did not get this from the documentation.
2) You can only log to the interpreter? Not for example to a file.

Thanks again for the quick answer!

**Comment by terrytangyuan** [comment]
1) The reason it's only happening during predict() in the example is that we are giving a `name` to a tensor that's only being returned in prediction mode - `mode_keys()$PREDICT` (see below). You can do something similar in other modes/phases as well. 
```
  predicted_classes <- tf$argmax(input = logits, axis = 1L, name = "predicted_classes")
  if (mode == mode_keys()$PREDICT) {
    predictions <- list(
      classes = predicted_classes,
      probabilities = tf$nn$softmax(logits, name = "softmax_tensor")
    )
    return(estimator_spec(mode = mode, predictions = predictions))
  }
```
2. Yes it only logs to the interpreter (at least the way it is currently). You can return those tensors you need from the list of `predictions` and then extract them if you want.
